src/pages/CookieClicker.vue

```diff
@@ -1,49 +1,331 @@
+}
+
+// Show toast message
+function show(message) {
+  toastMessage.value = message
+  showToast.value = true
+  setTimeout(() => {
+    showToast.value = false
+  }, 3000)
+}
+
+// Unlock achievements
+function checkAchievements() {
+  const unlocked = []
 
+  if (cookies.value >= 1000) unlocked.push("Cookie Enthusiast 🍪")
+  if (buildings.value.find(b => b.name === 'Grandma')?.count >= 10) unlocked.push("Grandma's Favorite 👵")
+  if (buildings.value.find(b => b.name === 'Factory')?.count >= 1) unlocked.push("Industrialist 🏭")
+
+  unlocked.forEach(name => {
+    if (!achievements.value.includes(name)) {
+      achievements.value.push(name)
+      show("🏆 Achievement unlocked: " + name)
+    }
+  })
+}
+
+// Unlock upgrades
+function checkForUpgrades() {
+  const cursor = buildings.value.find(b => b.name === 'Cursor')
+  if (cursor.count >= 5 && !upgrades.value.includes('Fast Fingers')) {
+    upgrades.value.push('Fast Fingers')
+    show("⚡ Upgrade unlocked: Fast Fingers (x2 CPS)")
+  }
+}
 </script>
+
 <template>
-    <div class="columns">
-        <div class="column is-4 has-background-primary has-text-centered">
-            <h1 class="is-size-1">{{ +cookies.toFixed(1) }} cookies</h1>
-            <h3 class="is-size-3">{{ +cps.toFixed(1) }} cps</h3>
-            <figure class="image is-square" @click="cookies++">
-                <img src="https://sweetlorens.com/cdn/shop/products/Copy-of-Chocolate-Chunk-Full-Cookie-transparent-background.png?v=1687811511" />
-            </figure>
-        </div>
-        <div class="column is-6 has-background-link">
-            asdas
-        </div>
-        <div class="column is-2 has-background-warning">
-            <button v-for="building in buildings" :disabled="cookies<building.price" @click="buyBuilding(building)" class="button is-primary is-large is-fullwidth">
-                {{ building.name }} 🍪{{ building.price }} #{{ building.count }}
-            </button>
+  <div class="columns is-gapless main-container">
+    <!-- Left Column -->
+    <div class="column is-4 has-text-centered p-5 left-panel">
+      <h1 class="is-size-1 has-text-white">{{ cookies.toFixed(1) }} cookies</h1>
+      <h3 class="is-size-3 has-text-light">{{ cps.toFixed(1) }} cps</h3>
+
+      <div class="cookie-container" @click="cookies++">
+        <img
+          class="cookie-img"
+          src="https://sweetlorens.com/cdn/shop/products/Copy-of-Chocolate-Chunk-Full-Cookie-transparent-background.png?v=1687811511"
+          alt="Main Cookie"
+        />
+      </div>
+    </div>
+
+    <!-- Middle Column -->
+    <div class="column is-6 middle-panel">
+      <div class="has-text-centered p-6 is-size-4 has-text-light">
+        <p>Welcome to Cookie Clicker Deluxe 🍪</p>
+        <p>Golden Cookies appear randomly – click fast!</p>
+
+        <div class="mt-6">
+          <h3 class="has-text-white is-size-5 mb-2">🏆 Achievements</h3>
+          <ul>
+            <li v-for="a in achievements" :key="a" class="has-text-warning">
```

Comment:
:key needs to be unique and mostly its fine what you are using but nothing is forcing this to be unique. If you are making it into object maybe add some id or something. You can look at our ToDo example